driver/docker/docker.go

```diff
@@ -293,9 +293,10 @@ func (d *Driver) exec(op *driver.Operation) (driver.OperationResult, error) {
 	return opResult, err
 }
 
-func (d *Driver) applyConfigurationOptions() error {
+// ApplyConfigurationOptions applies the configuration options set on the driver
+func (d *Driver) ApplyConfigurationOptions() error {
```

Comment:
Question: I wonder what is the use case in porter to export this function? As dockerConfigurationOptions is not exported so noone can add new options, this function shouldn't be called twice? Or maybe from an API point of view this function should then have a variadic parameter with new options to apply?

Reply:
The use case is to enable applying these configuration options outside of/without invoking the `exec` method where they otherwise will be applied.  Admittedly, this was originally motivated by unit testability, when we needed to check the config options that Porter adds without running exec (i.e. without actually invoking Docker).

However, I'd challenge the notion that this function shouldn't be called twice (or more). What would the fallout be if run multiple times?  The results of running `ApplyConfigurationOptions` should be idempotent unless the actual config option functions are updated (via the pre-existing `AddConfigurationOptions` method).  Speaking of which, you are right that `dockerConfigurationOptions` is not exported but API consumers can indeed add/modify options, and have been able to for some time, via the aforementioned [AddConfigurationOptions](https://github.com/cnabio/cnab-go/blob/master/driver/docker/docker.go#L50-L53)

Reply:
You are right I forgot this function! Ok to me then :+1: